src/main/java/org/osc/controller/nsfc/api/NeutronSfcSdnRedirectionApi.java

```diff
@@ -242,10 +254,35 @@ public InspectionHookElement getInspectionHook(String inspectionHookId) throws E
     // SFC methods
     @Override
     public NetworkElement registerNetworkElement(List<NetworkElement> portPairGroupList) throws Exception {
+        // NEW //////////////////////////////////////////
 
-        ServiceFunctionChainEntity sfc = new ServiceFunctionChainEntity();
         //check for null or empty list
         this.utils.throwExceptionIfNullOrEmptyNetworkElementList(portPairGroupList, "Port Pair Group member list");
+        this.utils.validatePPGList(portPairGroupList);
+
+        List<String> portPairGroupIds = portPairGroupList
+                                            .stream()
+                                            .map(ppg -> ppg.getElementId())
+                                            .collect(Collectors.toList());
+
+        PortChain portChain = Builders.portChain()
+                                    .description("Port Chain object created by OSC")
+                                    .chainParameters(emptyMap())
```

Comment:
are these required? if all we are passing are empty lists...why are these needed?